Add analyzer comments for elixir doc_spec_order common check (#2098)
exercism/website-copy: analyzer-comments/elixir/solution/doc_spec_order.md

# doc spec order

[comment]: # (requires 2 params:)
[comment]: # (- expected: How the doc and spec order should be)
[comment]: # (- actual: The incorrect order that the student used)

> There is not a correct order between the `@doc` and `@spec` modules attributes. But the elixir community convention is to use `@doc` first and `@spec` next to the function  

Reviewer: ```suggestion
Developers can choose the order of the `@doc` and `@spec` modules attributes, but the Elixir community convention is to use `@doc` first and `@spec` next to the function.
```
